release/nightly_tests/dataset/image_loader_microbenchmark.py

```diff
@@ -25,6 +26,219 @@
+                batch_size=args.batch_size,
+                feature_columns="image",
+                label_columns="label",
+            )
+            iterate(
+                ray_dataset.to_tf(
+                    batch_size=args.batch_size,
+                    feature_columns="image",
+                    label_columns="label",
+                ),
+                "ray_data_tfrecords+transform",
+                args.batch_size,
+                metrics,
+                args.output_file,
+            )
+
+    if args.parquet_data_root is not None:
+        ray_dataset = get_ray_parquet_dataset(args.parquet_data_root, parallelism=128)
+        for i in range(args.num_epochs):
+            iterate(
+                ray_dataset.iter_torch_batches(batch_size=args.batch_size),
+                "ray_data_parquet+map_transform",
+                args.batch_size,
+                metrics,
+                args.output_file,
+            )
+        print(ray_dataset.stats())
+
+    if args.mosaic_data_root is not None:
+        num_workers = None
+        mosaic_dl = get_mosaic_dataloader(
+            args.mosaic_data_root,
+            batch_size=args.batch_size,
+            num_physical_nodes=1,
+            num_workers=num_workers,
+        )
+
+        for i in range(args.num_epochs):
+            iterate(
+                mosaic_dl, "mosaicml_mds", args.batch_size, metrics, args.output_file
+            )
+
+        # ray.data.
+        use_s3 = args.mosaic_data_root.startswith("s3://")
+        if not use_s3:
+            mds_source = MdsDatasource(args.mosaic_data_root)
+            ray_dataset = ray.data.read_datasource(mds_source)
+            ray_dataset = ray_dataset.map(crop_and_flip_image)
+            for i in range(args.num_epochs):
+                iterate(
+                    ray_dataset.iter_torch_batches(batch_size=args.batch_size),
+                    "ray_data_mds+map_transform",
+                    args.batch_size,
+                    metrics,
+                    args.output_file,
+                )
+
+    return metrics
+
+
+def iterate(dataset, label, batch_size, metrics, output_file=None):
+    start = time.time()
+    it = iter(dataset)
+    num_rows = 0
+    print_at = 1000
+    for batch in it:
+        # note(swang): this will be slightly off if batch_size does not divide
+        # evenly into number of images but should be okay for large enough
+        # datasets.
+        num_rows += batch_size
+        if num_rows >= print_at:
+            print(f"Read {num_rows} rows")
+            print_at = ((num_rows // 1000) + 1) * 1000
+    end = time.time()
+    print(label, end - start)
+
+    tput = num_rows / (end - start)
+    print(label, "tput", tput)
+    metrics[label] = tput
+
+    if output_file is None:
+        output_file = "output.csv"
+    with open(output_file, "a+") as f:
+        for label, tput in metrics.items():
+            f.write(f"{label},{tput}\n")
```

Comment:
This code is taken from the earlier implementation of this release test: https://github.com/ray-project/ray/commit/c38ce4df20af537e06d17f65c04cf16efcd864fe#diff-0685e38f738e359aea5d19acfd692e0386c37b09b340a694cc3a9152c7b06441L28-L54